src/Mod/Path/PathScripts/PathPocketShapeGui.py

```diff
@@ -178,11 +178,13 @@ def cleanupPage(self, obj):
         # Unfortunately there's no direct way to determine the object's
         # livelihood without causing an error so we look for the object
         # in the document and clean up if it still exists.
-        for o in self.obj.Document.getObjectsByLabel(self.obj.Label):
-            if o == obj:
-                self.obj.ViewObject.RootNode.removeChild(self.switch)
-                return
-        PathLog.debug("%s already destroyed - no cleanup required" % (obj.Label))
+        try:
+            for o in self.obj.Document.getObjectsByLabel(self.obj.Label):
```

Comment:
I'm not sure this is the best way of doing it.

Can't you just test `self.obj` itself? If I create a document, then create an object, then delete the document I cannot access the `obj` any more, so testing this would seem easier.
```py
try:
    if hasattr(self.obj, "Name"):
        pass
except ReferenceError:
    PathLog.debug("obj already destroyed - no cleanup required")
    return

for o in self.obj.Document.getObjectsByLabel(self.obj.Label):
    if o == obj:
        self.obj.ViewObject.RootNode.removeChild(self.switch)
```

Also, getting the object by label seems weird, as if you can access the `self.obj` then you don't need to cycle through the document objects; you already have access to the object and to `self.switch`.
```py
try:
    if hasattr(self.obj, "Name"):
        pass
except ReferenceError:
    PathLog.debug("obj already destroyed - no cleanup required")
    return

self.obj.ViewObject.RootNode.removeChild(self.switch)
```

Reply:
That code was already there, not sure why it was written that way. I just added the exception handling to fix it. Your fix is a lot cleaner though, I've updated the pr.